types/umzug/index.d.ts

```diff
@@ -116,6 +118,27 @@ declare namespace umzug {
 
     }
 
+    interface MongoDBStorageOptions extends Storage {
+
+        /**
+         * The MongoDB database connection instance.
+         */
+        connection: MongoDB.Db;
```

Comment:
From https://github.com/sequelize/umzug/blob/master/src/storages/MongoDBStorage.js#L23 looks like connection is optional too (if collection is present)